### DISCUSSION TITLE
How to generate an ArgoCD token?

### DISCUSSION THREAD

**Discussion by cbwilliamsnh** [RoadieHQ/roadie-backstage-plugins#287] [Q&A] [answered]
I've installed both the frontend and backend Argo CD plugins (Option 2 for multiple instances) and I'm getting this error when trying to display the card:

`Error occurred while fetching data. SyntaxError: Unexpected token p in JSON at position 4`

I followed these steps to create my ArgoCD token: https://argoproj.github.io/argo-workflows/access-token/

What am I missing?

Thank you in advance,
Chuck

**Comment by Xantier** [comment]
The token should creatable from Argo CD UI directly.
There are few prerequisites to this:
* You need to be logged in as a user who has ability to create tokens/apikeys
* You should have a have user who has the ability to use apikeys 

If both of those are set up for your Argo CD instance and you navigate to `$YOUR_ARGO_INSTANCE_URL/settings/accounts/${THE_USER_YOU_WANT_TOKEN_TO_BE_CREATED}`, there should exist a button on the UI to generate a token.
![image](https://user-images.githubusercontent.com/2392775/142623839-182e9099-0281-49d5-afd8-9ea8bf4cfd79.png)

Note that the URL you linked to points to Argo Workflows, which would be a different product from the same Argo project.

**Reply by cbwilliamsnh** [reply]
@Xantier thank you! I have generated a proper token for a user, but I'm still seeing the error.

The token is set per the install instructions to:

`ARGO_CD_TOKEN='argocd.token=<MY TOKEN>'`

Thoughts?

**Comment by cbwilliamsnh** [comment, marked as answer]
I was missing the `/argo-cd` at the end of my URL in `app-config.yaml`! Working now!
